ferrocene/tools/self-test/src/binaries.rs

```diff
@@ -204,6 +204,7 @@ mod tests {
         }
     }
 
+    #[cfg(not(target_os = "macos"))] // On Darwin, empty executables exit with code 0.
```

Comment:
Can we instead write some other data to the file, causing macOS to fail? I'd prefer if we don't disable the test on that platform.

Reply:
As far as I can tell, anything I write there gets interpreted as a shell script, and will either exit with a status code. Eg writing `[]` results in a 0 exit. `[1,2,3]` results in an 32512 exit status. Writing `kill -9 $$` results in exit 9.